article/_posts/2019-07-24-Superhuman-AI-For-Multiplayer-Poker.md

```diff
@@ -0,0 +1,76 @@
+![](https://scontent.fyhu1-1.fna.fbcdn.net/v/t39.2365-6/67125564_680986665659447_1357790886280298496_n.gif?_nc_cat=103&_nc_oc=AQmje7JZHa4PZzM2z3k5YltkGUOxApzeFPMkw0Fp0QicfvLrzaLgcyg-1xMlX5XYUx4&_nc_ht=scontent.fyhu1-1.fna&oh=187da90f735c77a8b3e624bb7a211efd&oe=5DE92A5D)
+> source: facebook.com
+
+In this game, there are infinite ways the equilibrium can be reached. However, if every player computes this equilibrium separately, it is very unlikely the players will end up equally spaced. In six-player poker, the authors decided that the algorithm should try to find such equilibrium, and instead focus on creating an AI that beats humans empirically.
+
+## Hidden information
+
+A successful poker-playing AI must reason about the game's information and pick the optimal strategy. However, Poker involves bluffing, where the optimal strategy might not only depend on your cards. It is, therefore, necessary to balance the probability of your opponent having a strong hand versus them bluffing. Perfect information games can use tree search algorithms such as [Alpha-Beta Pruining](https://en.wikipedia.org/wiki/Alpha%E2%80%93beta_pruning) to estimate future action outcomes. In imperfect information games, the value of an action is dependant on the probability it would be taken. 
+
+Pluribus was trained via self-play, where the agent plays against copies of itself, with a modified version of [Monte-Carlo Counterfactual Regret Minimization](https://papers.nips.cc/paper/4569-efficient-monte-carlo-counterfactual-regret-minimization-in-games-with-many-player-actions.pdf) to update its policy. 
+
+[![](/article/images/pluribus/thumbnail.jpeg)](https://www.facebook.com/FacebookAI/videos/459177014638910/)
+
+Because the agent is playing against itself, it can know what would have happened if a different action had been chosen. As you can imagine, maintaining regret for every action in a game would be intractable. Therefore, the authors used "bucketing", where similar actions and states are grouped together. For example, raising 200$ or 201$ dollar will provide similar outcomes, and an opponent having a 10-straight or a 9-straight will usually act the same.
+
+The self-play training provides only a "blueprint" strategy that influences only the first betting round. Afterward, the agent uses MCCFRM to adapt its strategy to opponents. Moreso, tree search algorithms usually presume that the opponent will stick to a single strategy throughout the whole game, which is not something poker players do. Pluribus instead uses an approach where it will assume that each player may act according to different strategies beyond the leaf nodes. With Pluribus, the different strategies are variations of the blueprint, each more biased towards folding, calling or raising.
```

Comment:
I would present the MCCFRM acronym in the previous paragraph where you mention it.

Reply:
Fixed in cb5a976

Reply:
:+1: Thanks Antoine.